web-extensions/chrome/engine/utils.js

```diff
@@ -0,0 +1,66 @@
+/**
+ *
+ * The FrameFinder class is an utility class which finds a frame from the
+ * web-extension based on the provided frame Definition.
+ * All Frame definitions are static members of the class, and this serves as a
+ * central point to reference all frames handled by the web extension.
+ *
+ * @class
+ */
+function FrameFinder() {
+}
+
+FrameFinder.definitions = {
+    SIDEPANEL: {
+        id: 'BmcSidePanel',
+        inspect: frame => !!(frame.showHideSidePanel),
+    },
+};
+
+/**
+ * This static method returns the first frame matching the provided frame
+ * Definition object (which should be one of the frameDefinitions owned by the
+ * class).
+ *
+ * @param {Object} frameDef - the frame Definition as definied by FrameFinder
+ *                            class
+ *
+ * @return {Frame} the DOM frame object matching the definition provided
+ */
+FrameFinder.findWindow = function(frameDef) {
+    if (!frameDef) {
+        alert('Cannot find an frame with unknown Definition');
```

Comment:
`console.error`?

Reply:
I'll return false then (defaulting to "not found" behavior)

Reply:
errata, need to look over it, cause it's either a frame or undefined, but I need to ensure that undefined is handled well on top of it, otherwise it'll break the extension.